static/js/public/manage-snaps.js

```diff
@@ -0,0 +1,58 @@
+function getSelectedSnaps(snapCheckboxes) {
+  let snaps = [];
+
+  snapCheckboxes.forEach((snap) => {
+    if (snap.checked) {
+      snaps.push({
+        name: snap.dataset.snapName,
+      });
+    }
+  });
+
+  return snaps;
+}
+
+function updateSnaps() {
+  const manageSnapsForm = document.querySelector("#manage-snaps-form");
+  const saveButton = document.querySelector(".js-save-button");
+  const snapCheckboxes = Array.prototype.slice.call(
+    document.querySelectorAll(".js-snap-checkbox")
+  );
+
+  const originalSnapState = getSelectedSnaps(snapCheckboxes);
```

Comment:
Couldn't you just querySelectorAll with `:checked` to get the checked issue?

Reply:
Good point. I've had to scope it to the table though to be on the safe side.

Reply:
My question was can we replace formatSelectedSnaoData with a `querySelectorAll('.input:selected')`